Userland/Libraries/LibGUI/TreeView.cpp

```diff
@@ -480,6 +480,11 @@ void TreeView::move_cursor(CursorMovement movement, SelectionUpdate selection_up
 {
     auto& model = *this->model();
 
+    if (!cursor_index().is_valid()) {
+        set_cursor(model.index(0, model.tree_column(), cursor_index()), SelectionUpdate::Set);
+        return;
+    }
```

Comment:
This worked nicely, the return might seem excessive but it felt more natural. Now the first keypress selects the first row.

Reply:
This is already an improvement! If you press `[end]` you will still end up with the first item selected. If you'd also like to solve that, you would need to replace occurrences of `cursor_index()` in this method with a variable that you define here.

Reply:
In that case, let's remove the return statement and let it act on the first keypress :)